pennylane/hf/matrices.py

```diff
@@ -137,19 +137,17 @@ def kinetic(*args):
         """
         n = len(basis_functions)
         kinetic_matrix = anp.zeros((n, n))
-        for i, a in enumerate(basis_functions):
-            for j, b in enumerate(basis_functions):
-                if i <= j:
-                    if args:
-                        args_ab = []
-                        for arg in args:
-                            args_ab.append(arg[[i, j]])
-                        kinetic_integral = generate_kinetic(a, b)(*args_ab)
-                    else:
-                        kinetic_integral = generate_kinetic(a, b)()
-                    o = anp.zeros((n, n))
-                    o[i, j] = o[j, i] = 1.0
-                    kinetic_matrix = kinetic_matrix + kinetic_integral * o
+
+        for (i, a), (j, b) in it.combinations_with_replacement(enumerate(basis_functions), r=2):
```

Comment:
Just double checking: it contains the condition `if i < j`? If this condition is not there, the tests will still pass but we will be computing repetitive terms. So the efficiency will be affected. 

Why `combinations_with_replacement` is used here but not in the previous function?